salt/client/__init__.py

```diff
@@ -863,9 +882,9 @@ def get_iter_returns(
         # iterator for this job's return
         if self.opts['order_masters']:
             # If we are a MoM, we need to gather expected minions from downstreams masters.
-            ret_iter = self.get_returns_no_block(jid, tags_regex=['^syndic/.*/{0}'.format(jid)])
+            ret_iter = self.get_returns_short_block(jid, tags_regex=['^syndic/.*/{0}'.format(jid)], wait=timeout / 4)
         else:
-            ret_iter = self.get_returns_no_block(jid)
+            ret_iter = self.get_returns_short_block(jid, wait=timeout / 4)
```

Comment:
I played around with this a little bit. Unsetting the master `timeout` value so that it's back to the default causes the test suite to start, but it's about 3x slower. Setting it to 30s has the same effect.

The config files are in `salt.tests.integration.files.conf` if you'd like to experiment yourself.